src/GraphQL.Tests/Extensions/TypeExtensionTests.cs

```diff
@@ -0,0 +1,100 @@
+using GraphQL.Types;
+using GraphQL.Types.Relay;
+using GraphQL.Types.Relay.DataObjects;
+
+namespace GraphQL.Tests.Extensions;
+
+[Collection("StaticTests")]
+public class TypeExtensionTests
+{
+    [Theory]
+
+    // v8 naming
+    [InlineData(false, false, typeof(Class1), "Class1")]
+    [InlineData(false, false, typeof(Class1Type), "Class1")]
+    [InlineData(false, false, typeof(Class1Graph), "Class1Graph")]
+    [InlineData(false, false, typeof(Class1GraphType), "Class1")]
+    [InlineData(false, false, typeof(GenericClass<>), "TGenericClass")]
+    [InlineData(false, false, typeof(GenericClassType<>), "TGenericClass")]
+    [InlineData(false, false, typeof(GenericClassGraph<>), "TGenericClassGraph")]
+    [InlineData(false, false, typeof(GenericClassGraphType<>), "TGenericClass")]
+    [InlineData(false, false, typeof(GenericClass<Class1>), "Class1GenericClass")]
+    [InlineData(false, false, typeof(GenericClassType<Class1>), "Class1GenericClass")]
+    [InlineData(false, false, typeof(GenericClassGraph<Class1>), "Class1GenericClassGraph")]
+    [InlineData(false, false, typeof(GenericClassGraphType<Class1>), "Class1GenericClass")]
+    [InlineData(false, false, typeof(GenericClass<Class1Type>), "Class1GenericClass")]
+    [InlineData(false, false, typeof(GenericClass<Class1Graph>), "Class1GraphGenericClass")]
+    [InlineData(false, false, typeof(GenericClass<Class1GraphType>), "Class1GenericClass")]
+    [InlineData(false, false, typeof(GenericClass<Class1, string>), "Class1StringGenericClass")]
+    [InlineData(false, false, typeof(GenericClassType<Class1, string>), "Class1StringGenericClass")]
+    [InlineData(false, false, typeof(GenericClassGraph<Class1, string>), "Class1StringGenericClassGraph")]
+    [InlineData(false, false, typeof(GenericClassGraphType<Class1, string>), "Class1StringGenericClass")]
+    [InlineData(false, true, typeof(Class1), "TypeExtensionTests_Class1")]
+    [InlineData(false, true, typeof(GenericClass<Class1>), "TypeExtensionTests_Class1TypeExtensionTests_GenericClass")]
```

Comment:
Hmmm.....

Reply:
Yeah, this is ugly... Maybe the declaring type name should be applied only once? On the other hand, the generic class and generic argument can be declared in different parent classes, and users will expect to get different prefixes. Any combination of generic and nested classes seems to give an ugly solution. So, maybe just blame the users if they do that?

Reply:
Original issue was #2457 

I'm fine to leave it as shown here, letting users come up with their own solution if they run into problems.  `UseDeclaringTypeNames` always seemed to be a very edge-case fix; it doesn't help if the same class is in different namespaces, for instance.  I wouldn't have merged the feature to begin with, but it does not really hurt to leave it there.